kafka_consumer_freshness_tracker/src/main/java/com/tesla/data/consumer/freshness/ConsumerFreshness.java

```diff
@@ -113,22 +128,69 @@ public static void main(String[] args) throws IOException, InterruptedException
   }
 
   private void setup(Map<String, Object> conf) {
-    this.burrow = new Burrow((Map<String, Object>) conf.get("burrow"));
+    setupWithBurrow(conf, new Burrow((Map<String, Object>) conf.get("burrow")));
+  }
+
+  @VisibleForTesting
+  void setupWithBurrow(Map<String, Object> conf, Burrow burrow) {
+    this.burrow = burrow;
     int workerThreadCount = (int) conf.getOrDefault("workerThreadCount", DEFAULT_WORKER_THREADS);
     this.availableWorkers = ((List<Map<String, Object>>) conf.get("clusters")).stream()
         .map(clusterConf -> {
+          // validate the cluster configuration
+          Optional<String> validationErrorMsg = validateClusterConf(clusterConf);
+          if (validationErrorMsg.isPresent()) {
+            LOG.error("configuration for cluster " + (String) clusterConf.get("name") + " is invalid", validationErrorMsg);
+            return null;
+          }
           // allow each cluster to override the number of workers, if desired
           int numConsumers = (int) clusterConf.getOrDefault("numConsumers", DEFAULT_KAFKA_CONSUMER_COUNT);
           ArrayBlockingQueue<KafkaConsumer> queue = new ArrayBlockingQueue<>(numConsumers);
           for (int i = 0; i < numConsumers; i++) {
             queue.add(createConsumer(clusterConf));
           }
           return new AbstractMap.SimpleEntry<>((String) clusterConf.get("name"), queue);
-        }).collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+        })
+        .filter(cluster -> cluster != null)
+        .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
 
     this.executor = MoreExecutors.listeningDecorator(Executors.newFixedThreadPool(workerThreadCount));
   }
 
+  private Optional<String> validateClusterConf(Map<String, Object> clusterConf) {
+    final Map<String, Object> clusterDetail;
+    try {
+      clusterDetail = this.burrow.getClusterDetail((String) clusterConf.get("name"));
+    } catch (IOException e) {
+      String msg = "Failed to read cluster detail from burrow: " + e.getMessage();
+      LOG.error(msg);
+      this.metrics.burrowClusterDetailReadFailed.inc();
+      return Optional.of(msg);
+    }
+
+    final Map<String, Object> clusterDetailModuleSection = (Map<String, Object>) clusterDetail.get("module");
```

Comment:
This should be encapsulated in the `Burrow` class, which generally unpacks the burrow responses into 'domain specific' responses. For instance, i would expect to have a method like `Burrow#getClusterBootstrapServers` which just returned a list of strings of the bootstrap servers

Reply:
done